faker/providers/python/__init__.py

```diff
@@ -193,7 +194,7 @@ def _safe_random_int(self, min_value: float, max_value: float, positive: bool) -
             max_value = int(max_value - 1)
             if max_value < min_value:
                 max_value += 1
-            return self.random_int(min_value, max_value)
+            return self.random_float(min_value, max_value)
```

Comment:
`_safe_random_int` is designed to return an `int`, not a `float`. If you want to fix `pyfloat`, it will have to be fix at that level, or we'll have to refactor this method to be more generic.